frontend/src/components/widgets/EventsWidget.tsx

```diff
@@ -0,0 +1,154 @@
+                />
+              </tbody>
+            </table>
+            {data.length === 0 && <p>Aucun évenement à venir</p>}
+          </>
+        ) : null}
+      </Widget>
+      <EventModal event={selectedEvent} setEvent={setSelectedEvent} />
+    </>
+  );
+};
+
+const GroupEvents = ({ events, setSelectedEvent }) => {
+  const user = useContext(UserContext);
+
+  return events.reduce(
+    ({ body, currentDate }, event) => {
+      console.log(body, currentDate, event);
+      let eventDate = dayjs(event.startsAt).date();
+      let res = (
+        <>
+          {eventDate !== currentDate && (
+            <tr className={"border-0"}>
+              <td colSpan={2}>
+                <strong>
+                  {currentDate === dayjs().date()
+                    ? "Aujourd'hui"
+                    : dayjs(event.startsAt).date() ===
+                      dayjs().add(1, "day").date()
+                    ? "Demain"
+                    : "Le " + formatDate(event.startsAt)}
+                </strong>
+              </td>
+            </tr>
+          )}
+          <tr className={"border-0"}>
+            <td className={"py-0 align-middle"}>
+              <img
+                src={event.association.logo}
+                width={32}
+                alt={`Logo de ${event.association.name}`}
+              />{" "}
+              <button
+                className={"btn btn-link"}
+                onClick={(e) => {
+                  e.preventDefault();
+                  setSelectedEvent(event);
+                }}
+              >
+                {event.name}
+              </button>
+            </td>
+            <td className="py-0 text-right text-muted">
+              {event.participants.some(
+                (participant) => participant.id === user?.id
+              ) && <span className="tag tag-green mr-2">Inscrit</span>}
+              {formatTime(event.startsAt)}
+            </td>
+          </tr>
+        </>
+      );
+      return { body: [...body, res], currentDate: eventDate };
+    },
+    { body: [], currentDate: null }
+  ).body;
+};
+
+const EventModal = ({ event, setEvent }) => {
+  const user = useContext(UserContext);
+
+  return event != null ? (
+    <Modal show={true} onHide={() => setEvent(null)}>
+      <Modal.Body>
+        <h1>
+          {event.name}{" "}
+          {event.participants.some(
+            (participant) => participant.id === user?.id
+          ) && <span className="tag tag-green align-middle">Inscrit</span>}
+        </h1>
+        <p className="text-muted mb-2">
+          {event.place} | <EventDate event={event} />
+        </p>
+
+        <AvatarList className="mb-4">
+          {event.participants.map((user) => (
+            <UserAvatar
+              key={user.id}
+              userId={user.id}
+              size={Size.Medium}
+              link={false}
+              tooltip={`${user.firstName} ${user.lastName}`}
+            />
+          ))}
+        </AvatarList>
```

Comment:
The list is not limited, so this is going to blow the modal up if there are many participants
See in the event components how it's done, I think it's a better approach